src/Test/Utilities/Portable/Platform/CoreClr/TestExecutionLoadContext.cs

```diff
@@ -37,57 +36,22 @@ public TestExecutionLoadContext(IList<ModuleData> dependencies)
 
         protected override Assembly Load(AssemblyName assemblyName)
         {
-            lock (s_platformAssemblyPaths)
+            var comparer = StringComparer.OrdinalIgnoreCase;
+            var comparison = StringComparison.OrdinalIgnoreCase;
+            if (assemblyName.Name.StartsWith("System.", comparison) ||
+                assemblyName.Name.StartsWith("Microsoft.", comparison) ||
+                comparer.Equals(assemblyName.Name, "mscorlib") ||
+                comparer.Equals(assemblyName.Name, "netstandard"))
             {
-                Assembly assembly;
-                if (s_platformAssemblyPaths.TryGetValue(assemblyName.Name, out var assemblyPath))
-                {
-                    assembly = LoadPlatformAssembly(assemblyName, assemblyPath);
-                    if (assembly != null)
-                    {
-                        return assembly;
-                    }
-                }
-
-                if (_dependencies.TryGetValue(assemblyName.FullName, out var moduleData))
-                {
-                    return LoadImageAsAssembly(moduleData.Image);
-                }
-
                 return null;
             }
-        }
 
-        private Assembly LoadPlatformAssembly(AssemblyName assemblyName, string assemblyPath)
-        {
-            lock (s_loadedPlatformAssemblies)
+            if (_dependencies.TryGetValue(assemblyName.FullName, out var moduleData))
```

Comment:
Race condition?

Reply:
Actually I'm unsure if this code needs to handle re-entry or not. @elinor-fung do we need to worry about parallel calls in `AssemblyLoadContext.Load` or will the runtime serialize calls here?

Reply:
@agocke how is this a race condition? A `Dictionary<TKey, TValue>` is explicitly safe for concurrent reads.

Reply:
The runtime will not serialize calls here.

Whether or not this actually needs to handle re-entry depends on your usage. It would need to be handled if:
- you can call the `Load*` APIs on the ALC instance from multiple threads concurrently
- your `AssemblyLoadContext.Load` implementation itself calls `LoadFromAssemblyName` (`LoadFromStream` and `LoadFromAssemblyPath` should be fine)
  - a single `Load*` call will all be on one thread such that the assembly specified for load and its dependencies will be synchronously loaded on the same thread, but if the implementation of `AssemblyLoadContext.Load` does something like call `LoadFromAssemblyName` that would result in re-entry

Reply:
>  A Dictionary<TKey, TValue> is explicitly safe for concurrent reads.

That's actually awesome. I didn't know that.